packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -394,7 +415,7 @@ export class RunningSummarizer implements IDisposable {
     private trySummarize(
         reason: SummarizeReason,
         cancellationToken = this.cancellationToken): void {
-        if (this.summarizingLock !== undefined) {
+        if (this.summarizingLock !== undefined || this.refreshSummaryAckLock !== undefined) {
```

Comment:
I wonder why you are adding new lock, instead of using existing one? Is this mostly to assert that there could be only one lockedRefreshSummaryAckAction() call at a time?

Reply:
They have different purposes. The refreshSummaryAckLock is simply to make sure another summarizer is not enqueued or processed until the refresh is done.  Unless I'm missing something, the majority of the time the refresh will happen is while the summarizingLock is being used. Like we submit the summay + op + wait for the Ack to proceed and finish summarizing.

Reply:
Ah, so you allow refresh while summary is running, but not other way around, right?
Can you please cover these nuances in comments? It's not very clear as it stands

Reply:
will do.